src/v/config/node_config.cc

```diff
@@ -169,6 +172,27 @@ node_config::node_config() noexcept
       "injection",
       {.visibility = visibility::tunable},
       std::nullopt)
+  , debug_bundle_write_dir(
+      *this,
+      "debug_bundle_write_dir",
+      "The full path to the directory where debug bundle zip files are stored "
+      "(default: "
+      "/tmp)",
+      {.visibility = visibility::user},
+      "/tmp",
+      [](const std::filesystem::path& write_dir) -> std::optional<ss::sstring> {
+          if (!std::filesystem::exists(write_dir)) {
+              return ss::sstring{"Debug bundle write dir does not exist"};
+          }
+
+          // access is apart of unistd.h
+          if (access(write_dir.c_str(), R_OK | W_OK) != 0) {
+              return ss::sstring{"Debug bundle write dir does not have read "
+                                 "and write permissions"};
+          }
+
+          return std::nullopt;
```

Comment:
is this a reasonable thing to do in this validation? i'd have expected this to be done during start-up in application.cc somewhere.

Reply:
This is to guard against a user changing the directory to one where Redpanda can't write to after Redpanda starts

Reply:
do we have an async validation version?

Reply:
> do we have an async validation version?

IIUC, no we do not. The validator on `property<T>` and it a simple synchronous call that happens when the yaml config file is read.

> is this a reasonable thing to do in this validation

What is the risk of doing this validation in the config store?

Reply:
@michael-redpanda tested offline and it looks like the validators are not firing at all. I will test this out later after I get to the other comments.